src/compiler/types.ts

```diff
@@ -3508,7 +3508,8 @@ namespace ts {
         | ModuleDeclaration
         | TypeAliasDeclaration
         | InterfaceDeclaration
-        | EnumDeclaration;
+        | EnumDeclaration
+        | ExportAssignment;

```

Comment:
Why was this added here (and the empty case in `transformTopLevelDeclaration`)? Without it being added to the `isLateVisibilityPaintedStatement`, it will still fail to be printed late.